benchmark/profile_throughput.py

```diff
@@ -224,53 +237,85 @@ def process_request(self,
                 ])
 
 
-def main(dataset: str,
-         model_path: str,
-         concurrency: int = 64,
-         num_prompts: int = 2000,
-         tp: int = 1,
-         top_k: int = 1,
-         top_p: float = 1.0,
-         temperature: float = 1.0,
-         stream_output: bool = True,
-         csv: str = './profile_throughput.csv',
-         log_level: str = 'ERROR',
-         seed: int = 0):
-    """Benchmark the request throughput of lmdeploy in localhost.
-
-    Args:
-        dataset (str): Path to the dataset
-        model_path (str): Path to a model in localhost or a model_repo_id in huggingface.co
-        concurrency (int, optional): Number of working threads to process the sampled prompts.
-            Defaults to 64.
-        num_prompts (int, optional): Number of prompts to process. Defaults to 2000.
```

Comment:
Has this argument been removed?

Reply:
larger test dataset leads to better throughput

```
number of prompt tokens: 1239680
number of completion tokens: 1162390
token throughput (completion token): 2639.144 token/s
token throughput (prompt + completion token): 5453.771 token/s
RPS (request per second): 11.352 req/s
RPM (request per minute): 681.134 req/min
```

But I am not sure if this is cheating.

Reply:
It's not removed. The arguments parser is changed from Fire to ArgumentParser
```python
parser.add_argument('-n',
                        '--num-prompts',
                        type=int,
                        help='Number of prompts to process',
                        default=5000)
```
I don't think it's cheating when use large `num_prompts`, because the larger `num_prompts` is, the less tail effect it will be.
However, we should do a balance between the benchmark time and benchmark performance.